File: Reqnroll/Chapter14/14.5-DataTableAccessorPattern/After/WIMP.Specs/StepDefinitions/MenuAdminStepDefinitions.cs
using Reqnroll;

using WIMP.App.Services;
using WIMP.Specs.Support;

namespace WIMP.Specs.StepDefinitions;

[Binding]
public class MenuAdminStepDefinitions(MenuService menuService)
{
    [Given("the restaurant menu is")]
    public void GivenTheRestaurantMenuIs(List<MenuItemData> menuItems)
    {
        foreach (var menuItem in menuItems)
        {
            menuService.AddPizzaItem(
                menuItem.Name,
                menuItem.Price ?? DomainDefaults.MenuItemPrice,
                menuItem.Calories ?? DomainDefaults.MenuItemCalories,
                menuItem.Ingredients ?? DomainDefaults.MenuItemIngredients);
        }
    }
}

File: CucumberJVM/Chapter14/14.5-DataTableAccessorPattern/After/src/test/java/com/wimp/app/specs/stepdefinitions/MenuAdminStepDefinitions.java
package com.wimp.app.specs.stepdefinitions;

import com.wimp.app.services.MenuService;
import com.wimp.app.specs.support.MenuItemData;
import io.cucumber.java.en.*;

import java.util.List;

public class MenuAdminStepDefinitions {

    private final MenuService menuService;

    public MenuAdminStepDefinitions(MenuService menuService) {
        this.menuService = menuService;
    }

    @Given("the restaurant menu is")
    public void theRestaurantMenuIs(List<MenuItemData> menuItems) {
        for (var row : menuItems) {
            menuService.addMenuItem(row.name(), row.price(), row.calories(), row.ingredients());
        }
    }
}
